### ISSUE TITLE
`writeFile` API is not obeying to node API

### ISSUE THREAD

**Issue by samuelstroschein** [inlang/inlang#704]

**Comment by samuelstroschein** [comment]
Please release `@inlang/core` immediately after fixing this issue and update the plugin template to use the new `@inlang/core` version. 

To publish a new version: 

1. Create a PR 
2. Run `npx changeset` in the PR
3. Merge the PR
4. Merge the auto generated "versioning packages" pr.

**Comment by araknast** [comment]
Unlike `readFile`, the `[encoding` argument to `writeFile` defaults to `utf-8`](https://nodejs.org/api/fs.html#fspromiseswritefilefile-data-options),  so if I'm understanding correctly, we shouldn't need an api change, accepting `data` as a `utf-8` string is already the default.

I don't think we will need to handle an encoding argument at all actually since the node api intelligently detects if the argument to `writeFile` is a buffer and behaves appropriately. "The encoding option is ignored if `data` is a buffer." -- according to the node docs.

The only thing necessary would be changing the type of `data` from `string` to `string | Uint8array`. I've already implemented this in my rewrite, and can push it upstream if it's needed immediately.

**Comment by araknast** [comment]
> For the sake of DX and simplicity of implementation, would it be better to enforce the encoding flag and thereby explicitly handle the encoding and align `readFile` with `writeFile`?

I don't think so, we will need to start supporting returning `Uint8Array` from `readFile` at some point anyways, since isomorphic-git requires it. At the moment I've worked around the behavior by modifying isomorphic-git's [`FileSystem`](https://github.com/inlang/isomorphic-git/blob/a1cee20f114db570f5824f04301eb6f35ae8c891/src/models/FileSystem.js#LL100C1-L112C4), but in the long run I thing we should plan on making the encoding argument optional all around.

**Comment by samuelstroschein** [comment]
Alright. As long as the Filesystem interface is a strict subset of node fs:promises and no breaking changes are to be expected, everything should be fine. 

Closing this issue because "false alarm".
